packages/table/src/table.tsx

```diff
@@ -679,29 +786,122 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
+
+        this.leftQuadrantScrollElement.scrollTop = nextScrollTop;
+
+        this.handleBodyScroll(event);
+    }
+
+    private handleLeftQuadrantWheel = (event: React.WheelEvent<HTMLElement>) => {
+        const nextScrollTop = this.leftQuadrantScrollElement.scrollTop;
+        const nextScrollLeft = this.mainQuadrantScrollElement.scrollLeft + event.deltaX;
+
+        this.mainQuadrantScrollElement.scrollTop = nextScrollTop;
+        this.mainQuadrantScrollElement.scrollLeft = nextScrollLeft;
+
+        this.topQuadrantScrollElement.scrollLeft = nextScrollLeft;
+
+        this.handleBodyScroll(event);
+    }
+
+    private handleTopLeftQuadrantWheel = (event: React.WheelEvent<HTMLElement>) => {
+        const nextScrollTop = this.mainQuadrantScrollElement.scrollTop + event.deltaY;
+        const nextScrollLeft = this.mainQuadrantScrollElement.scrollLeft + event.deltaX;
+
+        this.mainQuadrantScrollElement.scrollTop = nextScrollTop;
+        this.mainQuadrantScrollElement.scrollLeft = nextScrollLeft;
+
+        this.leftQuadrantScrollElement.scrollTop = nextScrollTop;
+        this.topQuadrantScrollElement.scrollLeft = nextScrollLeft;
+
+        this.handleBodyScroll(event);
+    }
+
+    private renderMenu(refHandler: (ref: HTMLElement) => void) {
         const classes = classNames(Classes.TABLE_MENU, {
             [Classes.TABLE_SELECTION_ENABLED]: this.isSelectionModeEnabled(RegionCardinality.FULL_TABLE),
         });
         return (
             <div
                 className={classes}
-                ref={this.setMenuRef}
+                ref={refHandler}
                 onClick={this.selectAll}
             >
                 {this.maybeRenderRegions(this.styleMenuRegion)}
             </div>
         );
     }
 
-    private syncMenuWidth() {
-        const { menuElement, rowHeaderElement } = this;
+    private syncMenuElementWidths() {
+        this.syncMenuElementWidth(this.mainQuadrantMenuElement);
+        this.syncMenuElementWidth(this.topQuadrantMenuElement);
+        this.syncMenuElementWidth(this.leftQuadrantMenuElement);
+        this.syncMenuElementWidth(this.topLeftQuadrantMenuElement);
+    }
+
+    private syncMenuElementWidth(menuElement: HTMLElement) {
+        const { rowHeaderElement } = this;
         if (menuElement != null && rowHeaderElement != null) {
             const width = rowHeaderElement.getBoundingClientRect().width;
             menuElement.style.width = `${width}px`;
         }
     }
 
+    private syncQuadrantSizes() {
+        const {
+            mainQuadrantMenuElement,
+            mainQuadrantScrollElement,
+            topQuadrantElement,
+            leftQuadrantElement,
+            topLeftQuadrantElement,
+        } = this;
+
+        const frozenColumnsCumulativeWidth = this.grid.getCumulativeWidthAt(this.props.numFrozenColumns - 1);
+        const frozenRowsCumulativeHeight = this.grid.getCumulativeHeightAt(this.props.numFrozenRows - 1);
```

Comment:
This `- 1` is a _little_ unintuitive at first glance. Maybe create two variables `maxFrozenRowIndex` and `maxFrozenColumnIndex`?